pkg/controller/backup_controller.go

```diff
@@ -377,6 +384,48 @@ func (controller *backupController) runBackup(backup *api.Backup, bucket string)
 	return kerrors.NewAggregate(errs)
 }
 
+func (controller *backupController) UploadBackup(logger logrus.FieldLogger, objectStore cloudprovider.ObjectStore, bucket, backupName string, metadata, backup, log io.Reader) error {
```

Comment:
I wonder if it would make sense to extract this to `pkg/backup` so that it's part of the reusable lib?